modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/NumericFunctionsTypeCoercionTest.java

```diff
@@ -0,0 +1,595 @@
+
+        List<Matcher<RexNode>> args = List.of(ofTypeWithoutCast(pair.first()), ofTypeWithoutCast(pair.second()));
+        Matcher<RelNode> matcher = new FunctionCallMatcher(args).resultWillBe(pair.first());
+
+        assertPlan("SELECT ROUND(C1, C2) FROM T", schema, matcher::matches, List.of());
+    }
+
+    private static Stream<NumericPair> round2Args() {
+        List<NativeType> ints = List.of(NativeTypes.INT8, NativeTypes.INT16, NativeTypes.INT32, NativeTypes.INT64);
+
+        return Arrays.stream(NumericPair.values()).filter(p -> ints.contains(p.second()));
+    }
+
+    @ParameterizedTest
+    @MethodSource("numeric")
+    public void sign(NativeType type) throws Exception {
+        IgniteSchema schema = createSchemaWithSingleColumnTable(type);
+
+        List<Matcher<RexNode>> args = List.of(ofTypeWithoutCast(type));
+        Matcher<RelNode> matcher = new FunctionCallMatcher(args).resultWillBe(type);
+
+        assertPlan("SELECT SIGN(C1) FROM T", schema, matcher::matches, List.of());
+    }
+
+    @ParameterizedTest
+    @MethodSource("numeric")
+    public void sin(NativeType type) throws Exception {
+        IgniteSchema schema = createSchemaWithSingleColumnTable(type);
+
+        List<Matcher<RexNode>> args = List.of(ofTypeWithoutCast(type));
+        Matcher<RelNode> matcher = new FunctionCallMatcher(args).resultWillBe(NativeTypes.DOUBLE);
+
+        assertPlan("SELECT SIN(C1) FROM T", schema, matcher::matches, List.of());
+    }
+
+    @ParameterizedTest
+    @MethodSource("numeric")
+    public void sinh(NativeType type) throws Exception {
+        IgniteSchema schema = createSchemaWithSingleColumnTable(type);
+
+        List<Matcher<RexNode>> args = List.of(ofTypeWithoutCast(type));
+        Matcher<RelNode> matcher = new FunctionCallMatcher(args).resultWillBe(NativeTypes.DOUBLE);
+
+        assertPlan("SELECT SINH(C1) FROM T", schema, matcher::matches, List.of());
+    }
+
+    @ParameterizedTest
+    @MethodSource("numeric")
+    public void tan(NativeType type) throws Exception {
+        IgniteSchema schema = createSchemaWithSingleColumnTable(type);
+
+        List<Matcher<RexNode>> args = List.of(ofTypeWithoutCast(type));
+        Matcher<RelNode> matcher = new FunctionCallMatcher(args).resultWillBe(NativeTypes.DOUBLE);
+
+        assertPlan("SELECT TAN(C1) FROM T", schema, matcher::matches, List.of());
+    }
+
+    @ParameterizedTest
+    @MethodSource("numeric")
+    public void tanh(NativeType type) throws Exception {
+        IgniteSchema schema = createSchemaWithSingleColumnTable(type);
+
+        List<Matcher<RexNode>> args = List.of(ofTypeWithoutCast(type));
+        Matcher<RelNode> matcher = new FunctionCallMatcher(args).resultWillBe(NativeTypes.DOUBLE);
+
+        assertPlan("SELECT TANH(C1) FROM T", schema, matcher::matches, List.of());
+    }
+
+    @ParameterizedTest
+    @MethodSource("numeric")
+    public void truncate(NativeType type) throws Exception {
```

Comment:
looks like we need to add tests for CEILING and FLOOR also :sunglasses:

Reply:
Thanks. Fixed.